### PR TITLE
Deriving strategies

### LINKED ISSUE #43: Add `-Wmissing-deriving-strategies` to cabal ghc options
Also, add `DerivingStrategies` to the list of default extensions.

### PR DESCRIPTION
**Pull request by SanchayanMaity**
Resolves #43

### CODE REVIEW

File: membrain.cabal [hunk 1]
```diff
@@ -50,6 +50,7 @@ common common-options
                        -fhide-source-paths
                        -Wmissing-export-lists
                        -Wpartial-fields
+                       -Wmissing-deriving-strategies
```
Comment (vrom911) [inline membrain.cabal]: Could you please put it under the conditional, so it won't produce warning about unknown pragma for earlier GHCs. Like it's done here:
https://github.com/kowainik/tomland/blob/295c2b7580534303723206ba651557adff385cf4/tomland.cabal#L58-L60

Comment (SanchayanMaity) [reply membrain.cabal]: Thank you. Pushed a commit fixing this.

File: membrain.cabal [hunk 2]
```diff
@@ -50,6 +50,8 @@ common common-options
                        -fhide-source-paths
                        -Wmissing-export-lists
                        -Wpartial-fields
+if impl(ghc >= 8.8.1)
+  ghc-options:         -Wmissing-deriving-strategies
```
Comment (vrom911) [inline membrain.cabal]: Oh, I know why CI is failing. I see that indentation went incorrect here, so this conditional is applied to the rest of the stanza instead. You should move it 2 spaces right:

```
  if impl(ghc >= 8.8.1)
    ghc-options:       -Wmissing-deriving-strategies
```

Comment (SanchayanMaity) [reply membrain.cabal]: Yes, I saw that just now. Sorry about that. Fixed it.

File: membrain.cabal [hunk 3]
```diff
@@ -50,6 +50,8 @@ common common-options
                        -fhide-source-paths
                        -Wmissing-export-lists
                        -Wpartial-fields
+  if impl(ghc >= 8.8.1)
+    ghc-options:       -Wmissing-deriving-strategies
```
Comment (chshersh) [inline membrain.cabal:54]: @SanchayanMaity while we are on this, could you also please error on missing deriving strategies to make sure they are used? You can see how it's done here in `tomland`:

* https://github.com/kowainik/tomland/blob/ff5506eaa3c1a207f1514a56d12ef377b8cdfaa0/tomland.cabal#L60

Comment (SanchayanMaity) [reply membrain.cabal:54]: Pushed. Thank you.

**Review by chshersh — APPROVED** [review]
Thank you for this contribution! :heart:

**Review by vrom911 — APPROVED** [review]
Great stuff! Thank you :+1:

**Pull request merged**